### PR TITLE
Issue 442

### PR DESCRIPTION
**Pull request by mreboud**
Solves #452, thanks to @dvandyk 

Solves #442; `BR`, `F_L`, `F_H`, `A_FB` now refer to the CP-averaged quantities. The not averaged one are labeled `*_CP_specific`

Solves #456; `B->K^*gamma` isospin asymmetry is now defined via an expression observable.

### CODE REVIEW

**Comment by dvandyk** [comment]
I think we should use ``BR`` as the CP-averaged branching ratio, always, and provide ``A_CP`` in addition. If people want to extract a CP specific rate, they can do so manually.

Ditto for all the other observables. ``S_i`` are always CP-averaged, by definition. ``A_I`` are always CP-asymmetries, by definition. ``F_L``, ``A_FB``, and similar angular observables with well-defined quantum numbers should be CP averages, too, I think.

File: eos/rare-b-decays/b-to-kstar-gamma-bfs2004_TEST.cc [hunk 1]
```diff
@@ -226,23 +229,22 @@ class BToKstarGammaBobethCompatibilityTest :
...
-            // This line is commented out since it requires a fix, see Github issue #456
-            // observables.push_back(Observable::make("B->K^*gamma::A_I", p, k, o));
+            observables.push_back(Observable::make("B->K^*gamma::A_I", p, k, o));
 
             std::string filename(EOS_SRCDIR "/eos/rare-b-decays/exclusive-b-to-s-gamma_TEST-btokstargamma.data");
 #ifdef EOS_GENERATE_TEST_DATA
             {
                 std::cout << "-- GENERATING test case data for B->K^*gamma --" << std::endl;
-                RandomNumberGenerator rng;
+                gsl_rng * rng = gsl_rng_alloc(gsl_rng_taus);
```
Comment (dvandyk) [inline eos/rare-b-decays/b-to-kstar-gamma-bfs2004_TEST.cc:238]: ``gsl_rng_taus`` is deprecated in favour of ``gsl_rng_taus2`` according to the GSL manual.

File: eos/rare-b-decays/b-to-kstar-ll-gp2004_TEST.cc [hunk 2]
```diff
@@ -588,15 +591,15 @@ class BToKstarDileptonLowRecoilBobethCompatibilityTest :
 #ifdef EOS_GENERATE_TEST_DATA
             {
                 std::cout << "-- GENERATING test case data for B->K^*ll at low recoil --" << std::endl;
-                RandomNumberGenerator rng;
+                gsl_rng * rng = gsl_rng_alloc(gsl_rng_taus);
```
Comment (dvandyk) [inline eos/rare-b-decays/b-to-kstar-ll-gp2004_TEST.cc:594]: Ditto.

File: eos/rare-b-decays/b-to-k-ll-gp2004_TEST.cc [hunk 3]
```diff
@@ -254,15 +260,15 @@ class BToKDileptonLowRecoilBobethCompatibilityTest :
 #ifdef EOS_GENERATE_TEST_DATA
             {
                 std::cout << "-- GENERATING test case data for B->Kll at low recoil --" << std::endl;
-                RandomNumberGenerator rng;
+                gsl_rng * rng = gsl_rng_alloc(gsl_rng_taus);
```
Comment (dvandyk) [inline eos/rare-b-decays/b-to-k-ll-gp2004_TEST.cc:263]: See below.

**Review by dvandyk — COMMENTED** [review]
I approve in principle, with only a comment on the use of ``gsl_rng_taus``.

The test case failure is truly puzzling, needs to be understood first.

File: eos/rare-b-decays/b-to-kstar-gamma-bfs2004_TEST.cc (hunk 1 above)
Comment (mreboud) [reply eos/rare-b-decays/b-to-kstar-gamma-bfs2004_TEST.cc]: Good point, I missed that

**Pull request merged**